src/cromshell/submit/command.py

```diff
@@ -0,0 +1,131 @@
+@click.argument("dependencies_zip", type=click.Path(exists=True),  required=False)
+@click.pass_obj
+def main(config, wdl, wdl_json, options_json, dependencies_zip):
+    """Submit a workflow and arguments to the Cromwell Server"""
+
+    LOGGER.info("submit")
+
+    io_utils.assert_required_file_is_not_empty(wdl, "WDL")
+    io_utils.assert_required_file_is_not_empty(wdl_json, "Input JSON")
+    io_utils.assert_required_file_is_not_empty(options_json, "Options json")
+    io_utils.assert_required_file_is_not_empty(dependencies_zip, "Dependencies Zip")
+
+    # At this point, we should validate our inputs if we can:
+    womtool_validate_wdl_and_json(wdl, wdl_json)
+
+    # Check if Cromwell Server Backend works
+    http_utils.assert_can_communicate_with_server(config)
+
+    # Submit workflow to Cromwell Server
+    LOGGER.info(f"Submitting job to server: {cromshellconfig.cromwell_server}")
+    submission_params = {
+        'workflowSource': open(wdl, 'rb'),
+        'workflowInputs': open(wdl_json, 'rb')
+    }
+    if options_json is not None:
+        submission_params['workflowOptions'] = open(options_json, 'rb')
+    if dependencies_zip is not None:
+        submission_params['workflowDependencies'] = open(dependencies_zip, 'rb')
+
+    # TODO: In order for this to work the ending "/" needs be removed from config.api_string
+    # TODO: This might need to be implemented in another PR because it involves editing
+    # TODO: the status command and http_util script
+    request_out = requests.post(f"{config.cromwell_server}{config.api_string}",
+                                files=submission_params)
+
+    # Check to make sure that we actually submitted the job correctly
+    # 1. Check for any initial failure by server to accept the job.
+    http_utils.check_http_request_status_code(request_out)
+
+    # 2. Check messages from server for workflow problems.
+    # Get our workflow status and job ID:
+    workflow_status = json.loads(request_out.content)
+
+    # 2.A If the status is not `Submitted`, something went wrong: Todo: did a manual test but not one with failed workflow
+    if workflow_status['status'] != "Submitted":
+        log.display_logo(io_utils.dead_turtle)
+        LOGGER.error("Error: Server reports that job was not properly submitted.")
+        LOGGER.error(f"Server response:{request_out.text}")
+        raise Exception("Error: Server reports that job was not properly submitted.\n"
+            f"Server response:{request_out.text}")
+
+    # 2.B If the ID is not an ID, something went wrong:
+    if io_utils.validate_workflow_id(workflow_status['id']) != 0:
+        log.display_logo(io_utils.dead_turtle)
+        LOGGER.error("Error: Did not get a valid ID back. Something went wrong.")
+        LOGGER.error(f"Server response:{request_out.text}")
+        raise Exception("Error: Did not get a valid ID back. Something went wrong.\n"
+            f"Server response:{request_out.text}")
```

Comment:
this block of code can be refactored into another function.

Reply:
done